PR: https://github.com/odoo/odoo/pull/223960

From: 5715844d093dcd7687677e80664ded6099d125f0
From: Mélanie
Date: 2025-08-25 05:59:26

Breaking data model changes scores: del:2.4 + add:0, change matches:
-    work_location_name = fields.Char(compute="_compute_work_location_name")
+    work_location_name = fields.Char(related='employee_id.work_location_name')
-    work_location_name = fields.Char("Work Location Name", compute="_compute_work_location_name_type")
-    work_location_type = fields.Selection([

Total Changes: 43

[FIX] hr,hr_holidays,hr_homeworking : missing selection values in hr_icon_display

Step to reproduce:
*-----------------
    * Go on Employee
    * Select one employee
    * Add as today worklocation "office"
    * Hover over hr presence icon

The title will be "Building 1, second floor" and not "Office"

task-4933881

closes odoo/odoo#223960

X-original-commit: 0496ed10636c7b2dfde7038a43494d4edbd9f95b
Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr/models/hr_employee.py
+++ b/addons/hr/models/hr_employee.py
@@ -141,6 +141,11 @@ class HrEmployee(models.Model):
     study_school = fields.Char("School", groups="hr.group_hr_user", tracking=True)
     emergency_contact = fields.Char(groups="hr.group_hr_user", tracking=True)
     emergency_phone = fields.Char(groups="hr.group_hr_user", tracking=True)
+    work_location_name = fields.Char("Work Location Name", compute="_compute_work_location_name")
+    work_location_type = fields.Selection([
+        ("home", "Home"),
+        ("office", "Office"),
+        ("other", "Other")], compute="_compute_work_location_type", tracking=True)
 
     # employee in company
     parent_id = fields.Many2one('hr.employee', 'Manager', tracking=True, index=True,
@@ -334,6 +339,16 @@ class HrEmployee(models.Model):
                 version = employee.current_version_id
             employee.version_id = version
 
+    @api.depends("version_id.work_location_id.name")
+    def _compute_work_location_name(self):
+        for employee in self:
+            employee.work_location_name = employee.version_id.work_location_id.name or None
+
+    @api.depends("version_id.work_location_id.location_type")
+    def _compute_work_location_type(self):
+        for employee in self:
+            employee.work_location_type = employee.version_id.work_location_id.location_type or 'other'
+
     @api.depends('version_ids.date_version', 'version_ids.active', 'active')
     def _compute_current_version_id(self):
         for employee in self:

--- a/addons/hr/models/hr_employee_public.py
+++ b/addons/hr/models/hr_employee_public.py
@@ -33,7 +33,7 @@ class HrEmployeePublic(models.Model):
     email = fields.Char(related='employee_id.email')
     work_contact_id = fields.Many2one('res.partner', readonly=True)
     work_location_id = fields.Many2one('hr.work.location', readonly=True)
-    work_location_name = fields.Char(compute="_compute_work_location_name")
+    work_location_name = fields.Char(related='employee_id.work_location_name')
     work_location_type = fields.Selection(related='employee_id.work_location_type')
     user_id = fields.Many2one('res.users', readonly=True)
     resource_id = fields.Many2one('resource.resource', readonly=True)
@@ -44,12 +44,9 @@ class HrEmployeePublic(models.Model):
         ('absent', 'Absent'),
         ('archive', 'Archived'),
         ('out_of_working_hour', 'Out of Working hours')], compute='_compute_presence_state', default='out_of_working_hour')
-    hr_icon_display = fields.Selection([
-        ('presence_present', 'Present'),
-        ('presence_out_of_working_hour', 'Out of Working hours'),
-        ('presence_absent', 'Absent'),
-        ('presence_archive', 'Archived'),
-        ('presence_undetermined', 'Undetermined')], compute='_compute_presence_icon')
+    hr_icon_display = fields.Selection(
+        selection='_get_selection_hr_icon_display',
+        compute='_compute_presence_icon')
     show_hr_icon_display = fields.Boolean(compute='_compute_presence_icon')
     last_activity = fields.Date(compute="_compute_last_activity")
     last_activity_time = fields.Char(compute="_compute_last_activity")
@@ -81,6 +78,9 @@ class HrEmployeePublic(models.Model):
     # Crap
     newly_hired = fields.Boolean('Newly Hired', compute='_compute_newly_hired', search='_search_newly_hired')
 
+    def _get_selection_hr_icon_display(self):
+        return self.env['hr.employee']._fields['hr_icon_display']._description_selection(self.env)
+
     def _compute_from_employee(self, field_names):
         if isinstance(field_names, str):
             field_names = [field_names]
@@ -133,9 +133,6 @@ class HrEmployeePublic(models.Model):
     def _compute_member_of_department(self):
         self._compute_from_employee('member_of_department')
 
-    def _compute_work_location_name(self):
-        self._compute_from_employee('work_location_name')
-
     def _get_manager_only_fields(self):
         return []
 

--- a/addons/hr/models/hr_version.py
+++ b/addons/hr/models/hr_version.py
@@ -130,11 +130,6 @@ class HrVersion(models.Model):
         tracking=True)
     work_location_id = fields.Many2one('hr.work.location', 'Work Location',
                                        domain="[('address_id', '=', address_id)]", tracking=True)
-    work_location_name = fields.Char("Work Location Name", compute="_compute_work_location_name_type")
-    work_location_type = fields.Selection([
-        ("home", "Home"),
-        ("office", "Office"),
-        ("other", "Other")], compute="_compute_work_location_name_type", tracking=True)
 
     departure_reason_id = fields.Many2one("hr.departure.reason", string="Departure Reason",
                                           groups="hr.group_hr_user", copy=False, ondelete='restrict', tracking=True)
@@ -511,12 +506,6 @@ class HrVersion(models.Model):
             if not version.structure_type_id or (version.structure_type_id.country_id and version.structure_type_id.country_id != version.company_id.country_id):
                 version.structure_type_id = _default_salary_structure(version.company_id.country_id.id)
 
-    @api.depends("work_location_id.name", "work_location_id.location_type")
-    def _compute_work_location_name_type(self):
-        for version in self:
-            version.work_location_name = version.work_location_id.name or None
-            version.work_location_type = version.work_location_id.location_type or 'other'
-
     @api.depends('distance_home_work', 'distance_home_work_unit')
     def _compute_km_home_work(self):
         for version in self:
